### ISSUE TITLE
Bug: Script is minimized even in dev mode

### ISSUE THREAD

**Issue by filipac** [roots/bud#1211]
### Agreement

- [X] This is not a duplicate of an existing issue
- [X] I have read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

### Expected Behavior

I've tested https://github.com/roots/bud/pull/1182 and works ok now but I found another issue/bug:

The app.js script is minified even in development mode and that breaks react-refresh as it does not know what to re-render.

If I add this code to bud config, it works and the js file is not minified:

```.tap(bud => {
      if (!app.isProduction) {
        bud.config((c) => {
          c.optimization.minimize = false
          return c
        })
      }
    })
```

### Actual Behavior

The final build is minified then running yarn dev

### Steps To Reproduce

1. Run yarn.dev
2. Check out app.js file and see that it is minified

### version

5.4.0

### What package manager are you using?

yarn classic

### version

1.22.17

### Are you using pnpm?

- [ ] yes
- [X] no

**Comment by kellymears** [comment]
Thanks for the report, I snuck a fix for this into #1213. It's caused by the CLI overwriting the option.

Your override works because that function wraps a hook that gets called very late.

**Comment by filipac** [comment]
nice, keep up the good work!

**Comment by kellymears** [comment]
I think this can be closed now that 5.5.0 is released.
